core/lib/spree/app_configuration.rb

```diff
@@ -145,6 +145,10 @@ class AppConfiguration < Preferences::Configuration
     #   @return [String] Two-letter ISO code of a {Spree::Country} to assumed as the country of an unidentified customer (default: "US")
     preference :default_country_iso, :string, default: 'US'
 
+    # @!attribute [rw] default_email_regexp
+    #   @return [Regexp] Regex to be used in email validations, for example in Spree::EmailValidator
+    preference :default_email_regexp, :regexp, default: /\A([^@\.]|[^@\.]([^@\s]*)[^@\.])@([^@\s]+\.)+[^@\s]+\z/
```

Comment:
Might be worth trying with Ruby's built in [URI::MailTo::EMAIL_REGEXP](https://github.com/ruby/ruby/blob/48b94b791997881929c739c64f95ac30f3fd0bb9/lib/uri/mailto.rb#L55)

Reply:
This is just a copy of the regex in Spree::EmailValidator. That regex you mentioned is significantly different from it and I think changing it might have a potential impact for unaware upgraders.

Reply:
I think it might be worth considering an update of the regex because the current one has a couple of bugs.

[At this link](https://regex101.com/r/JTDNrI/4) you can see and play with different combinations (be aware of the usage of `^` and `$` instead of `\A` and `\z`, it was just to use multiline matches), but looks like that the current regex allows the following:

- `test @example.com`: single whitespace before the `@`
- ` test@example.com`: leading whitespace, although this is a smaller issue since Devise by default auto strips leading and trailing whitespaces on `email` fields. The `Spree::User` class in `solidus_auth_devise` is immune from this bug because of the auto stripping.

Since this would enter a deprecation cycle, would it be ok updating the regex to something like this? [Resulting in this example](https://regex101.com/r/3rC6us/1).

```ruby
/\A([^@\s.]|[^@\s.]([^@\s]*)[^@\s.])@([^@\s]+\.)+[^@\s]+\z/
```

@cesartalves , @kennyadsl , @softr8 thoughts?

Reply:
I see no problem in updating the regex. We can let users know that it's gonna happen on a release note. Besides, shouldn't break a lot of validations save few edge cases (e.g., white spaces as you mentioned).